src/daft-connect/src/spark_analyzer.rs

```diff
@@ -88,7 +90,11 @@ impl SparkAnalyzer<'_> {
         })
     }
 
-    pub async fn to_logical_plan(&self, relation: Relation) -> ConnectResult<LogicalPlanBuilder> {
+    #[allow(clippy::wrong_self_convention)]
+    pub async fn to_logical_plan(
+        &mut self,
```

Comment:
What's the purpose of taking `&mut self`? I don't think the struct is mutated anywhere.

Reply:
oh good catch. I needed mutability for my original implementation.